e2e-tests/src/driver.rs

```diff
@@ -83,7 +84,27 @@ impl Driver {
 
         let mut sandbox_environments: Vec<ProtocolSandboxEnvironment> = Vec::default();
         for protocol_sandbox in &self.config.protocol_sandboxes {
+            let protocol_name = match protocol_sandbox {
+                ProtocolSandboxConfig::Stellar(_) => "stellar",
+                ProtocolSandboxConfig::Near(_) => "near",
+                ProtocolSandboxConfig::Icp(_) => "icp",
+            };
+
+            if !self.environment.protocols.is_empty()
+                && !self
+                    .environment
+                    .protocols
+                    .iter()
+                    .any(|p| p.to_lowercase() == protocol_name)
```

Comment:
[an empty iterator returns false](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any)

```suggestion
            if !self
                .environment
                .protocols
                .iter()
                .any(|p| p.to_lowercase() == protocol_name)
```